### PR TITLE
[Code Review - Do Not Merge]  

### PR DESCRIPTION
**Pull request by idcmp**

### CODE REVIEW

File: build/depends.py [hunk 1]
```diff
@@ -1236,6 +1236,8 @@ def sources(self, build):
                    "util/indexrange.cpp",
                    "util/widgetrendertimer.cpp",
 
+                   "idcmp/junction.cpp",
```
Comment (daschuer) [inline build/depends.py]: You should find a reasonable folder name for your feature, related to the feature itself.

File: src/idcmp/junction.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#include "junction.h"
+
+IdcmpJunction::IdcmpJunction(PlayerManager *pPlayerManager)
+        : QObject(),
+          m_pPlayerManager(pPlayerManager) {
+
+    for (unsigned int deckNum = 1; deckNum <= pPlayerManager->numberOfDecks(); deckNum++) {
```
Comment (daschuer) [inline src/idcmp/junction.cpp]: We have also the [Master] num_decks you can use.

**Review by daschuer — CHANGES_REQUESTED** [review]
Nice! Do you have an idea how to make it usable for all users?

File: src/mixxx.h [hunk 3]
```diff
@@ -165,6 +167,8 @@ class MixxxMainWindow : public QMainWindow {
 #endif
     ControllerManager* m_pControllerManager;
 
+    IdcmpJunction *m_pIdcmpJunction;
```
Comment (daschuer) [inline src/mixxx.h]: We alligne the * with the type.

File: src/idcmp/junction.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    QFile file("/sys/kernel/debug/ec/ec0/io");
+
+    if (!file.open(QIODevice::ReadWrite)) {
+        qWarning() << "Could not open file " << file.fileName() << " error code: " << file.error();
+        return;
+    }
+
+    file.seek(12);
+    // thinklight off or thinklight blink
+    file.putChar((v == 0) ? 0x0a : 0xca);
```
Comment (daschuer) [inline src/idcmp/junction.cpp]: Here you should use the chapter litheral
'\r' what is 0xca?

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/idcmp/junction.cpp (hunk 4 above)
Comment (idcmp) [reply src/idcmp/junction.cpp]: Thanks for taking the time to review this!  

It's actually two nibbles. The first specifies either on/off/blink, the second specifies which light. More details here: https://www.reddit.com/r/thinkpad/comments/7n8eyu/thinkpad_led_control_under_gnulinux/

**Comment by Holzhaus** [comment]
I have not the slightest idea what this does. Close?

**Comment by uklotzde** [comment]
@idcmp Please open a new PR with a meaningful title and description when ready. Closing.

**Pull request closed without merging**